docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md

```diff
@@ -148,6 +148,121 @@ In the default OMS configuration, seven days are allocated to Stripe to complete
 
 For instructions on using payment details, like the payment reference, from Stripe, see [Retrieve and use payment details from third-party PSPs](https://docs.spryker.com/docs/pbc/all/payment-service-provider/{{page.version}}/base-shop/retrieve-and-use-payment-details-from-third-party-psps.html)
 
+## Embed the Stripe payment page using iframe
+
+By default, the Stripe App payment flow assumes that the payment page is on another domain. When a user places an order, they're redirected to the Stripe payment page. To improve the user experience, you can embed the Stripe payment page directly into your website as follows:
+
+
+1. Create or update `PaymentConfig.php` (`Pyz\Zed\Payment\PaymentConfig`) on the project level with the following configuration:
+```php
+namespace Pyz\Zed\Payment;
+
+class PaymentConfig extends \Spryker\Zed\Payment\PaymentConfig
+{
+    public function getStoreFrontPaymentPage(): string
+    {        
+        // Please make sure that domain is whitelisted in the config_default.php `$config[KernelConstants::DOMAIN_WHITELIST]`
+        return '/payment'; //or any other URL on your storefront domain e.g. https://your-site.com/payment-with-stripe
+    }
+}
+```
+
+In this setup, the redirect URL will be added as a `url` query parameter to the URL you've specified in the `getStoreFrontPaymentPage()` method; the value of the parameter is base64-encoded.
+
+
+2. Create a page to render the Stripe payment page. Here's a minimal PHP example:
+
+```php
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Order payment page</title>
+</head>
+<body>
+<iframe src="<?php echo base64_decode($_GET['url']) ?>" width="100%" height="700px" style="border:0"></iframe>
+</body>
+</html>
+```
+
+If your Storefront is based on third-party frontend technologies, follow the documentation of your framework to create a page to render the Stripe payment page using query parameters from the redirect URL provided in the Glue API `POST /checkout` response.
```

Comment:
If I'm using a third-party frontend, do I need to do steps 3-6?

Reply:
no, not needed, it's only for Yves